openfisca_web_api/openAPI.yml

```diff
@@ -1,374 +1,441 @@
+          type: "object"
+        description:
+          type: "string"
+        id:
+          type: "integer"
+          format: "string"
+        source:
+          type: "string"
+      example: null
+
+    Parameters:
+      type: "object"
+      additionalProperties:
+        type: "object"
+        properties:
+          description:
+            type: "string"
+          href:
+            type: "string"
+
+    Variable:
+      type: "object"
+      properties:
+        defaultValue:
+          type: "string"
+        definitionPeriod:
+          type: string
+          enum:
+            - MONTH
+            - YEAR
+            - ETERNITY
+        description:
+          type: "string"
+        entity:
+          type: "string"
+        formulas:
+          type: "object"
+          additionalProperties:
+            $ref: "#/components/schemas/Formula"
+        id:
+          type: "string"
+        reference:
+          type: "array"
+          items:
+            type: "string"
+        source:
+          type: "string"
+        valueType:
+          type: "string"
+          enum:
+            - Int
+            - Float
+            - Boolean
+            - Date
+            - String
+      example: null
+
+    Variables:
+      type: "object"
+      additionalProperties:
+        type: "object"
+        properties:
+          description:
+            type: "string"
+          href:
+            type: "string"
+
+    Formula:
+      type: "object"
+      properties:
+        content:
+          type: "string"
+        source:
+          type: "string"
+
+    Brackets:
+      type: "object"
+      additionalProperties:
+        type: "number"
+        format: "float"
+
+    Values:
+      description: All keys are ISO dates. Values can be numbers, booleans, or arrays of a single type (number, boolean or string).
+      type: "object"
+      additionalProperties:
+        $ref: "#/components/schemas/Value"
+  #    propertyNames:  # this keyword is part of JSON Schema but is not supported in OpenAPI Specification at the time of writing, see https://swagger.io/docs/specification/data-models/keywords/#unsupported
+  #      pattern: "^[12][0-9]{3}-[01][0-9]-[0-3][0-9]$"  # all keys are ISO dates
+
+    Value:
+      oneOf:
+      - type: "boolean"
+      - type: "number"
+        format: "float"
+      - type: "array"
+        items:
+          oneOf:
+          - type: "string"
+          - type: "number"
+
+    Entities:
+      type: "object"
+      properties:
+        description:
+          type: "string"
+        documentation:
+          type: "string"
+        plural:
+          type: "string"
+        roles:
+          type: "object"
+          additionalProperties:
+            $ref: "#/components/schemas/Roles"
+    Roles:
+      type: "object"
+      properties:
+        description:
+          type: "string"
+        max:
+          type: "integer"
+        plural:
+          type: "string"
+    SituationInput: null
```

Comment:
Add a generic object with a description explaining these resources are generated dynamically.

This has two benefits:

1. Static validation
2. Make explicit the behaviour for modellers.

Reply:
So why isn't this done? 🙂